pkg/rancher-desktop/config/settings.ts

```diff
@@ -474,114 +473,142 @@ function updateSettings(settings: Settings) {
 // Imported from dashboard/config/settings.js
 // Setting IDs
 export const SETTING = { PL_RANCHER_VALUE: 'rancher' };
+const REGISTRY_PATH_PROFILE = ['SOFTWARE', 'Rancher Desktop', 'Profile'];
 
 /**
- * Read and validate the json formatted deployment profile.
- * Returns the system profile if it exists, otherwise the user profile if it exists.
- * @returns A settings profile stripped of all fields not found in the schema.
+ * Read and validate deployment profiles, giving system level profiles
+ * priority over user level profiles.  If the system directory contains a
+ * defaults or locked profile, the user directory will not be read.
+ * @returns undefined, or type validated defaults and locked deployment profiles.
  */
-export function readDeploymentProfile() {
-  let profileData;
-  let profile;
+export function readDeploymentProfiles() {
+  let defaults;
+  let locked;
 
   switch (os.platform()) {
   case 'win32':
-    try {
-      profile = readRegistryDeploymentProfile(); // TODO: call with path to system profile
-    } catch {
-      try {
-        profile = readRegistryDeploymentProfile(); // TODO: call with path to user profile
-      } catch {}
+    for (const key of [nativeReg.HKLM, nativeReg.HKCU]) {
+      const registryKey = nativeReg.openKey(key, REGISTRY_PATH_PROFILE.join('\\'), nativeReg.Access.READ);
+
+      if (registryKey !== null) {
```

Comment:
This `if` block should probably be in a  `try/catch` (with a `console.error` message so we can see what happened).

Reply:
done.